### PR TITLE
#163557852 add redirection to signup and signin pages

### PR DESCRIPTION
**Pull request by Danielshow**
#### What does this PR do?
This PR passes user data to all route when the user is logged in. It also redirects a user to the landing page if they are logged in and try accessing login or signup page.

#### Description of Task to be completed?
- pass user data as props to all route if a user is logged in
- redirect if user data exist when user visit signup and sign-in pages

#### How should this be manually tested?
N/A

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#163557852](https://www.pivotaltracker.com/story/show/163557852)

#### Screenshots (if appropriate)
N/A

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -2,5 +2,8 @@ import React from 'react';
 import { render } from 'react-dom';
 import App from './containers/App';
 import './assets/scss/base.scss';
+import decodeToken from './helpers/validationHelpers/decodeToken';
 
-render(<App />, document.querySelector('#root'));
+const user = decodeToken() ? decodeToken() : null;
```
Comment (efalayi) [inline src/index.js]: @Danielshow I don't get what's going on here.

Comment (Danielshow) [reply src/index.js]: @ef, I have fixed it, just checked now and `decodedToken` returns null if the token does not exist.

File: src/containers/App.js [hunk 2]
```diff
@@ -1,23 +1,27 @@
...
+          <Route
+            exact path='/'
+            render={props => <Home {...props} user={user} />}
+          />
         {routes.map(route => (
             <Route
             path={route.path}
             key={route.name}
-            component={route.component}
+            render={props => <route.component {...props} user={user} />}
```
Comment (efalayi) [inline src/containers/App.js]: 👍

File: src/containers/Article/__tests__/Article.spec.js [hunk 3]
```diff
@@ -38,8 +34,10 @@ describe('Article Component Test', () => {
         },
       },
     },
-    location: {
-      pathname: '/jdakdjkajda',
+    history: {
+      location: {
+        pathname: '/jdakdjkajda',
```
Comment (efalayi) [inline src/containers/Article/__tests__/Article.spec.js]: @Danielshow what does `jdakdjkajda` mean?

Comment (Danielshow) [reply src/containers/Article/__tests__/Article.spec.js]: That is just a random word, the pathname is getting called in the component so we set it as props. I will change the name if that's what you mean.

**Pull request merged**